app/controllers/confirmations_controller.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# Create Confirmations for Users
+class ConfirmationsController < Devise::ConfirmationsController
+  respond_to :json
+
+  def new
+    super
+  end
+
+  def create
+    super
+    @front_end_path = concatenate_path
+  end
+
+  def show
+    self.resource = resource_class.confirm_by_token(params[:confirmation_token])
+    if resource.errors.empty?
+      sign_in_resource(resource)
+    else
+      errors(resource.errors.details)
+      render json: error_response, status: :forbidden
+    end
+  end
+
+  private
+
+  def concatenate_path
+    options = ActionMailer::Base.default_url_options
+    "#{options[:protocol]}#{options[:host]}#{options[:port]}"
+  end
+
+  def errors(details = nil)
+    @errors ||= details
+  end
+
+  def error_response
+    {
+      error: I18n.t('errors.messages.confirmation'),
```

Comment:
This is not using the right locale.
1. sign up in English.
2. (before using a confirmation URL with a bad token)  Go to the login page and switch the language to Spanish.
3. Use a URL with a bad confirmation token
4. The alert displayed does not translate the error message:

<img width="432" alt="confirmation-error-wrong-locale-i18n" src="https://user-images.githubusercontent.com/673794/92013783-1454ad80-ed03-11ea-8248-52f5ae518750.png">


The backend (Rails) does not have any info about the locale so uses the default one (`:en`).
The frontend (React) just displays the message without doing any translation.
In this case, the front end _has_ to be responsible for doing the translation since Rails does not have any response header with the locale info prior to the GET (`show` with the `ConfirmationsController`)

(This will be true for any page displayed as the result of a link, yes?)

(I think this is a separate issue/PR but YMMV)

Reply:
@weedySeaDragon Can you write this ticket up, really great catch.

Reply:
HOLY CRAP - never mind!  I just found Github's "reference in a new issue" feature and I made the ticket!